chef/automate
# Adding date filter to GetNodesCounts API

### :nut_and_bolt: Description: What code changed, and why?
Adding date filters to the GetNodesCounts allows a user to filter on nodes that have check-in between a specific period. This was added for the insight UI control for the Desktop view. 

### :chains: Related Resources

https://github.com/chef/automate/issues/3372

### :athletic_shoe: How to Build and Test the Change
1. `build components/config-mgmt-service && build components/automate-gateway && start_all_services`
1. Add some nodes with decreasing check-in dates with this command `for i in {1..5}; do generate_chef_run_example | jq --arg d $(date +%FT%TZ -d "$i day ago") '.end_time=$d' | send_chef_data_raw; done`
1. To get the node counts for three days ago to a day ago run `curl -f --insecure -H "api-token: $(get_api_token)" --data-urlencode "start=$(date +%FT%TZ -d '3 day ago')" --data-urlencode "end=$(date +%FT%TZ -d '1 day ago')" https://localhost/api/v0/cfgmgmt/stats/node_counts | jq`
1. Ensure the response looks like below. 
```
{
  "total": 2,
  "success": 0,
  "failure": 0,
  "missing": 2
}
```

### :white_check_mark: Checklist
